Document intellij, and update lsp related docs (#1197)
returntocorp/semgrep-docs: docs/extensions/overview.md

### Official extensions

- Microsoft Visual Studio Code: [`semgrep-vscode`](https://marketplace.visualstudio.com/items?itemName=semgrep.semgrep)
- IntelliJ Ultimate Idea (and most other IntelliJ products) [`semgrep-intellij`](https://plugins.jetbrains.com/plugin/22622-semgrep)
- Emacs: [`lsp-mode`](https://github.com/emacs-lsp/lsp-mode)

### Community-contributed extensions

- Vim: [`semgrep.vim`](https://github.com/returntocorp/semgrep.vim)

### The LSP Command

> All of our official extensions use the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) to communicate

Reviewer: ```suggestion
All Semgrep official extensions use the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/) to communicate
```

> with Semgrep. This allows us to focus on one codebase that can be shared across most modern editor platforms. To implement a custom extension,

Reviewer: ```suggestion
with Semgrep. This allows the Semgrep team to focus on one codebase that can be shared across most modern editor platforms. To implement a custom extension,
```

> one can wrap `semgrep lsp` to start the Semgrep Language Server, which will communicate over `stdio`. Alternatively, this protocol is a 

Reviewer: ```suggestion
you can wrap `semgrep lsp` to start the Semgrep Language Server, which communicates over `stdio`. Alternatively, this protocol is a 
```

great way to integrate Semgrep into a project, as it can perform incremental scans, and caches various computations to hugely increase performance. 

> Please let us know on our community Slack linked below if you do, we'd be more than happy to help in anyway.

Reviewer: ```suggestion
Please let the Semgrep team know through the community Slack linked below if you do [what?], we'd be more than happy to help in anyway.
```

There's a do... without an action in this sentence.
